pom.xml

```diff
@@ -80,6 +80,26 @@
             </activation>
             <build>
                 <plugins>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>properties-maven-plugin</artifactId>
+                        <version>1.0.0</version>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>set-system-properties</goal>
+                                </goals>
+                                <configuration>
+                                    <properties>
+                                        <property>
+                                            <name>org.jenkinsci.plugins.durabletask.BourneShellScriptTest.TEST_BINARY</name>
+                                            <value>true</value>
+                                        </property>
```

Comment:
Ok, looking at this in more detail, I think that the way you were using the environment variable in the test previously made more sense. I don't really see a compelling reason why we'd want to generate binaries but not test them, so I think makes sense for them to be controlled by the same setting.

It looks like we have to use an environment variable to activate `generate-binaries-profile` because there is no way to pass arbitrary Java options to `buildPlugin`, so you are using `withEnv`. If that were not the case, I would say that the profile should be activated by system property and then we'd just use that property in the test.

So in summary I think it would be best to drop `BourneShellScriptTest.TEST_BINARY` here and just look at the environment variable in `BourneShellScriptTest` like you had before. Maybe change the name from `SKIP_BINARY_GENERATION` to `SKIP_DURABLE_TASK_BINARY_GENERATION` or something for clarity if the variable is set in a different context like the PCT.